core/src/main/java/org/wildfly/plugin/core/DeploymentOperations.java

```diff
@@ -0,0 +1,479 @@
+     * indicate a  wildcard for the address.
+     * </p>
+     *
+     * @param pairs the key/value pairs to use
+     *
+     * @return an address for the key/value pairs
+     */
+    public static ModelNode createAddress(final String... pairs) {
+        return createAddress(Arrays.asList(pairs));
+    }
+
+    /**
+     * Creates an {@linkplain ModelNode address} that can be used as the address for an operation. The address is
+     * simply a {@link ModelNode} of type {@link ModelType#LIST}.
+     * <p>
+     * The string is split into key/value pairs. If the final key does not have a value an {@code *} is used to
+     * indicate a  wildcard for the address.
+     * </p>
+     *
+     * @param pairs the key/value pairs to use
+     *
+     * @return an address for the key/value pairs
+     */
+    public static ModelNode createAddress(final Iterable<String> pairs) {
+        final ModelNode address = new ModelNode();
+        final Iterator<String> iterator = pairs.iterator();
+        while (iterator.hasNext()) {
+            final String key = iterator.next();
+            final String value = (iterator.hasNext() ? iterator.next() : "*");
+            address.add(key, value);
+        }
+        return address;
+    }
+
+    /**
+     * Creates an operation to add deployment content to the deployment repository.
+     * <p>
+     * Note this does not deploy the content.
+     * </p>
+     *
+     * @param content     the path to the content to deploy
+     * @param name        the name for the deployment content, if {@code null} the paths file name is used
+     * @param runtimeName the runtime name for the deployment content
+     *
+     * @return the add operation
+     */
+    public static Operation createDeploymentAddOperation(final Path content, final String name, final String runtimeName) {
+        Assertions.requiresNotNullParameter(content, "content");
+        final ModelNode address = createAddress(DEPLOYMENT, name);
+        final ModelNode addOperation = createAddOperation(address);
+        if (runtimeName != null) {
+            addOperation.get(RUNTIME_NAME).set(runtimeName);
+        }
+        final CompositeOperationBuilder builder = CompositeOperationBuilder.create(true);
+        addContent(builder, addOperation, content);
+        builder.addStep(addOperation);
+        return builder.build();
+    }
+
+    /**
+     * Creates an operation to add deployment content to the deployment repository.
+     * <p>
+     * Note this does not deploy the content.
+     * </p>
+     *
+     * @param content     the content to deploy
+     * @param name        the name for the deployment content
+     * @param runtimeName the runtime name for the deployment content, can be {@code null}
+     *
+     * @return the add operation
+     */
+    public static Operation createDeploymentAddOperation(final InputStream content, final String name, final String runtimeName) {
```

Comment:
Do we need this one, given that we have `createDeployOperation(final Deployment deployment)` ?